rmgweb/database/views.py

```diff
@@ -590,6 +1039,284 @@ def parseSoluteDataComment(comment):
+
+    # get the solvation properties if the input solvent was given
+    solvent_info = None
+    if solvent != 'None':
+        solute_data_results[f'dGsolv298({energy_unit})'] = []  # solvation free energy
+        solute_data_results[f'dHsolv298({energy_unit})'] = []  # solvation enthalpy
+        solute_data_results[f'dSsolv298({energy_unit}/K)'] = []  # solvation entropy
+
+        additional_info_list.append('dGsolv298: solvation free energy at 298 K.')
+        additional_info_list.append('dHsolv298: solvation enthalpy at 298 K.')
+        additional_info_list.append('dSsolv298: solvation entropy at 298 K.')
+
+        # get the solvent data and href
+        solvent_entry = database.solvation.libraries['solvent'].entries[solvent]
+        solvent_data = solvent_entry.data
+        solvent_smiles_list = []
+        for spc in solvent_entry.item:
+            solvent_smiles_list.append(spc.smiles)
+        solvent_index = solvent_entry.index
+        solvent_href = reverse('database:solvation-entry',
+                               kwargs={'section': 'libraries', 'subsection': 'solvent',
+                                       'index': solvent_index})
+        solvent_info = (solvent, solvent_smiles_list, solvent_href, solvent_index)
+
+        # check whether we have all solvent parameters to calculate dGsolv and dHsolv
+        abraham_parameter_list = [solvent_data.s_g, solvent_data.b_g, solvent_data.e_g, solvent_data.l_g,
+                                  solvent_data.a_g, solvent_data.c_g]
+        dGsolv_availble = not any(param is None for param in abraham_parameter_list)
+        mintz_parameter_list = [solvent_data.s_h, solvent_data.b_h, solvent_data.e_h, solvent_data.l_h,
+                                solvent_data.a_h, solvent_data.c_h]
+        dHsolv_availble = not any(param is None for param in mintz_parameter_list)
+
+        for solute_data, solute_data_found in zip(solute_data_list, solute_data_found_list):
+            dGsolv298 = '-'
+            dHsolv298 = '-'
+            dSsolv298 = '-'
+            if solute_data_found:
+                if dGsolv_availble:
+                    dGsolv298 = db.calc_g(solute_data, solvent_data) / 4184  # convert to kcal/mol
+                    if energy_unit == 'kJ/mol':
+                        dGsolv298 = dGsolv298 * 4.184  # convert to kJ/mol
+                else:
+                    dGsolv298 = 'not available'
+                if dHsolv_availble:
+                    dHsolv298 = db.calc_h(solute_data, solvent_data) / 4184  # convert to kJ/mol
+                    if energy_unit == 'kJ/mol':
+                        dHsolv298 = dHsolv298 * 4.184  # convert to kJ/mol
+                else:
+                    dHsolv298 = 'not available'
+                if dGsolv_availble and dHsolv_availble:
+                    dSsolv298 = (dHsolv298 - dGsolv298) / 298  # has the unit of dGsolv298 divided by Kelvin
+                    if abs(dSsolv298) < 1000:
```

Comment:
Maybe you can write a function that generally formats a number using these repeated criteria:
- if <1000 then display normally, otherwise use sci notation
- maybe divide by specified T (default = 298)
- maybe do unit conversions

Then you can just call that function each time instead of repeating them here.

Reply:
I created a new function `clean_up_value` for this now.